tools/jdbc/src/jni/duckdb_java.cpp

```diff
@@ -653,7 +664,8 @@ static std::string type_to_jduckdb_type(LogicalType logical_type) {
 		}
 	} break;
 	default:
-		return logical_type.ToString();
+		// this intentionally ignores aliases, so we know how to load the data on the java side
+		return EnumUtil::ToString(logical_type.id());
```

Comment:
It seems like this will cause the underlying (internal) type of a type to be loaded into the client, correct? I think this is actually not desirable - since the underlying internal type is mostly an implementation detail. For example, consider the [`INET` type in the `inet` extension](https://github.com/duckdb/duckdb/blob/main/extension/inet/inet_extension.cpp). Internally this type is stored as `STRUCT(ip_type UINT8, address HUGEINT, mask INT16)`. However, for display purposes you would want to display the actual IP and not reveal the internal details of how the type is stored, e.g. this is what the shell does:

```cpp
D load inet;
D create table inet(ip inet);
D insert into inet values ('127.0.0.1');
D from inet;
┌───────────┐
│    ip     │
│   inet    │
├───────────┤
│ 127.0.0.1 │
└───────────┘
```

This can be accomplished by using the to string cast that is supplied by the `inet` extension.

Reply:
If we're happy with that, and assuming that all extension types can be cast to a string, sure?

@Maxxen I'm assuming that will be okay for the GEOMETRY type for example?